### PR TITLE
Correct paths for hexagon-unknown-none-elf platform doc

### PR DESCRIPTION
**Pull request by androm3da**
Update the library paths to correctly refer to libdemo{1,2}_hexagon and switch to the release build instead.

Update references to libstandalone to the specific G0/PIC archive instead.

### CODE REVIEW

**Review by wesleywiser — APPROVED** [review]

**Comment by androm3da** [comment]
Thanks @wesleywiser !

After this is merged to `master` can I cherry-pick it somehow to the 1.76 branch?  I think this is `beta`?  The [contributing guide](https://rustc-dev-guide.rust-lang.org/contributing.html) is not exactly explicit about how to do this.  I'll just make a new PR like this one, but targeting `beta`?

**Comment by wesleywiser** [comment]
@androm3da We just need to tag it as `beta-nominated` and then the compiler team will decide whether or not to backport it to the beta branch (if that happens, someone from the release team will take care of it). 

Could you provide a bit of rationale why it's important this is backported to the 1.76 beta instead of riding the trains in 1.77? (Since this is a doc only change, this doesn't need to be super in-depth but we typically explain why a backport is being requested.)

**Comment by androm3da** [comment]
> @androm3da We just need to tag it as `beta-nominated` and then the compiler team will decide whether or not to backport it to the beta branch (if that happens, someone from the release team will take care of it).
> 
> Could you provide a bit of rationale why it's important this is backported to the 1.76 beta instead of riding the trains in 1.77? (Since this is a doc only change, this doesn't need to be super in-depth but we typically explain why a backport is being requested.)

The extent of the rationale is that "it was wrong when initially submitted (IIRC this was before the branch)".  Following the procedure from the baseline will result in confusing link errors.

It's probably not critical to backport, but I'd prefer it if possible.

**Comment by wesleywiser** [comment]
Thanks! It looks like https://github.com/rust-lang/rust/pull/117601 landed in 1.77 (current nightly) so depending on how quickly this PR is merged, it might not need a backport at all.

**Comment by apiraino** [comment]
Beta backport discussed but decision post-poned to next week ([Zulip discussion](https://rust-lang.zulipchat.com/#narrow/stream/238009-t-compiler.2Fmeetings/topic/.5Bweekly.5D.202024-02-01/near/419269985))

**Comment by androm3da** [comment]
> Thanks! It looks like #117601 landed in 1.77 (current nightly) so depending on how quickly this PR is merged, it might not need a backport at all.

Okay, my mistake.  I was confused about the timing of the branches and made assumptions based on the 1.75 release date announcement that it was to be included in 1.76.

**Comment by androm3da** [comment]
> Beta backport discussed but decision postponed to next week ([Zulip discussion](https://rust-lang.zulipchat.com/#narrow/stream/238009-t-compiler.2Fmeetings/topic/.5Bweekly.5D.202024-02-01/near/419269985))

Sorry, @apiraino I don't think it makes sense to backport this without #117601 and since 117601 was a feature it probably isn't a candidate to backport.

I'll remove the `beta-nominated` tag.

**Comment by apiraino** [comment]
@androm3da no worries and thanks for the additional context. I'll remove the backport nomination

@rustbot label -beta-nominated

**Comment by apiraino** [comment]
Beta backport accepted as per compiler team [on Zulip](https://rust-lang.zulipchat.com/#narrow/stream/238009-t-compiler.2Fmeetings/topic/.5Bweekly.5D.202024-02-08/near/420479645)

Going to backport it anyway because it's a nice and not risky. The labels on this PR and the intersection with #117601 were a bit confusing. #117601 is in 1.77 (now beta) but this PR merged too late and is in 1.78 (nightly) so it didn't make it to the beta. Backporting this will result having in the curent beta both #117601 and the platform documentation fix.

@rustbot label +beta-nominated +beta-accepted

**Pull request merged**
